assets/js/util/partition-report.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Report partitioning utility.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Splits report data into segments for current and comparison ranges.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Array}  report           Report rows.
+ * @param {Object} args             Additional arguments.
+ * @param {number} args.rangeLength Length of an individual range.
+ * @return {Object} Object with keys for `compareRange` and `currentRange`.
+ */
+export const partitionReport = ( report, { rangeLength } ) => {
+	invariant( Array.isArray( report ), 'report must be an array to partition.' );
+	invariant( Number.isInteger( rangeLength ) && rangeLength > 0, 'rangeLength must be a positive integer.' );
+
+	const compareRange = report.slice( 0, rangeLength );
+	const currentRange = report.slice( rangeLength, rangeLength * 2 );
+	// Normalize the length of both ranges to be the same.
+	const minLength = Math.min( compareRange.length, currentRange.length );
+
+	return {
+		compareRange: compareRange.slice( 0, minLength ),
+		currentRange: currentRange.slice( 0, minLength ),
```

Comment:
Could this result in a potential problem that the compare date range would be incorrect (e.g. missing some days)? For example, if the `currentRange` is 1 entry longer than `compareRange`, right now this would strip the last result of the `compareRange`, so effectively that date would be skipped which would result in somewhat of a wrong comparison (where furthermore even for a "multitude of a week" period weekdays may not be aligned).

A more appropriate solution may be to slice the `compareRange` from its end, effectively trimming away the oldest dates. This may be more specific to Site Kit's behavior, but I think that makes sense - we don't need this utility to be overly generic.

Reply:
> Could this result in a potential problem that the compare date range would be incorrect (e.g. missing some days)?

I don't think so – it shouldn't be possible for the `compareRange` to be shorter than the `currentRange` since it is sliced from the beginning of the `report` array. Due to the way the two ranges are sliced before this, `currentRange` can never be longer because they both have the same length limit.

I suppose this makes the slicing on the `compareRange` here redundant and we could simply slice the `currentRange` to the length of the `compareRange` instead.